internal/cmd/command.go

```diff
@@ -0,0 +1,275 @@
+		RunE:         handleReadCommand,
+		SilenceUsage: true,
+	}
+	readCmd.Flags().StringVarP(&deviceName, "device", "d", "", "Specify the name of device")
+	readCmd.Flags().StringVarP(&commandName, "command", "c", "", "Specify the name of the command to be executed")
+	readCmd.Flags().BoolVarP(&pushEvent, "pushevent", "p", false, "If set, a successful GET will result in an event being pushed to the EdgeX system")
+	readCmd.Flags().BoolVarP(&noReturnEvent, "noreturnevent", "r", false, "If set, there will be no Event returned in the http response")
+	readCmd.MarkFlagRequired("device")
+	readCmd.MarkFlagRequired("command")
+	cmd.AddCommand(readCmd)
+	addFormatFlags(readCmd)
+}
+
+func initWriteCommand(cmd *cobra.Command) {
+	var writeCmd = &cobra.Command{
+		Use:          "write",
+		Short:        "write command referenced by the command name and device name",
+		Long:         `Issue the specified write command referenced by the command name to the device/sensor that is also referenced by name`,
+		RunE:         handleWriteCommand,
+		SilenceUsage: true,
+	}
+	writeCmd.Flags().StringVarP(&deviceName, "device", "d", "", "Specify the name of the device")
+	writeCmd.Flags().StringVarP(&commandName, "command", "c", "", "Specify the name of the command to be executed")
+	writeCmd.Flags().StringVarP(&requestBody, "body", "b", "", "Specify PUT requests body/data")
+	writeCmd.Flags().StringVarP(&requestFile, "file", "f", "", "Specify a file containing PUT requests body/data")
+	writeCmd.MarkFlagRequired("device")
+	writeCmd.MarkFlagRequired("command")
+	cmd.AddCommand(writeCmd)
+	addFormatFlags(writeCmd)
+}
+
+func initListCommand(cmd *cobra.Command) {
+	listCmd := &cobra.Command{
+		Use:          "list",
+		Short:        "A list of device supported commands",
+		Long:         "Returns a paginated list contains all of the commands in the system associated with their respective device, optionally filtering by device name",
+		RunE:         handleListCommand,
+		SilenceUsage: true,
+	}
+	listCmd.Flags().StringVarP(&deviceName, "device", "d", "", "List commands specified by device name")
+	listCmd.Flags().IntVarP(&limit, "limit", "l", 50, "The number of items to return. Specifying -1 will return all remaining items")
+	listCmd.Flags().IntVarP(&offset, "offset", "o", 0, "The number of items to skip")
+	cmd.AddCommand(listCmd)
+	addFormatFlags(listCmd)
+}
+
+func handleReadCommand(cmd *cobra.Command, args []string) error {
+	dsPushEvent := boolToString(pushEvent)
+	dsReturnEvent := boolToString(!noReturnEvent)
+
+	//issue READ command and handle output if nothing can be displayed
+	response, err := getCoreCommandService().IssueReadCommand(deviceName, commandName, dsPushEvent, dsReturnEvent)
+	if err != nil {
+		return err
+	}
+
+	if response == nil {
+		fmt.Println("Read command issued. Nothing to display. Please retry without flag -r.")
```

Comment:
Shouldn't we catch this before sending the command? I think this probably a bug in the Core Command code too, as you a READ command with `--noreturnevent` set, nothing will happen because `--pushevent` defaults to false.

Reply:
You're right, when `--noreturnevent` set, and `--pushevent` default to false, there is nothing to do. I move this part before sending the command and print the message: "Nothing to do. Please remove -noreturnevent flag or set -pushevent flag."